packages/protocol/test/AccountRegistry/AccountRegistryManager.js

```diff
@@ -0,0 +1,233 @@
+                await manager.deployProxy(initialBehaviourAddress, aceAddress, trustedGSNSignerAddress);
+
+                const postDeployEpoch = await manager.latestEpoch();
+                expect(postDeployEpoch.toString()).to.equal('1');
+            });
+        });
+
+        describe('Upgrade pattern', async () => {
+            let behaviour;
+            let manager;
+
+            beforeEach(async () => {
+                behaviour = await Behaviour1.new();
+                manager = await AccountRegistryManager.new(opts);
+
+                const initialBehaviourAddress = behaviour.address;
+                await manager.deployProxy(initialBehaviourAddress, aceAddress, trustedGSNSignerAddress, opts);
+            });
+
+            it('should successfully register account mapping', async () => {
+                // register an account with the extension, which will be put in storage under accountMapping
+                const { privateKey, address } = secp256k1.generateAccount();
+                const linkedPublicKey = keccak256('0x01');
+                const spendingPublicKey = keccak256('0x0');
+
+                const domain = signer.generateAccountRegistryDomainParams(behaviour.address);
+                const message = {
+                    account: address,
+                    linkedPublicKey,
+                };
+
+                const encodedTypedData = typedData.encodeTypedData({
+                    domain,
+                    ...ACCOUNT_REGISTRY_SIGNATURE,
+                    message,
+                });
+
+                const signature = secp256k1.ecdsa.signMessage(encodedTypedData, privateKey);
+
+                const r = signature[1];
+                const s = signature[2].slice(2);
+                const v = signature[0].slice(-2);
+                const sig = r + s + v;
+
+                await behaviour.registerAZTECExtension(address, linkedPublicKey, spendingPublicKey, sig);
+
+                const storedLinkedPublicKey = await behaviour.accountMapping.call(address);
+                expect(storedLinkedPublicKey).to.equal(linkedPublicKey);
+            });
+
+            it('should upgrade to a new Account Registry behaviour contract', async () => {
+                const testBehaviour = await TestBehaviour.new();
+                const existingProxy = await manager.proxy.call();
+
+                await manager.upgradeAccountRegistry(existingProxy, testBehaviour.address);
+
+                const postUpgradeProxy = await manager.proxy.call();
+                expect(postUpgradeProxy).to.equal(existingProxy);
+
+                const newBehaviourAddress = await manager.getImplementation.call(existingProxy);
+                const expectedNewBehaviourAddress = testBehaviour.address;
+                expect(newBehaviourAddress).to.equal(expectedNewBehaviourAddress);
+
+                const isNewFeatureImplemented = 'newFeature()' in testBehaviour.methods;
```

Comment:
This just tests that the contract you're using as a behaviour has the new method you put on that same contract. Would be nice to see some calls to the proxy which more explicitly test that upgrades/initialisations have happened correctly.

Reply:
Yep agreed. Have now adapted that test to call a `newFeature()` function on the upgraded behaviour contract via the `proxyContract`